app/api/routes/resource_modification.py

```diff
@@ -49,8 +53,19 @@ def update_resource(id, json, db):
             resource.languages = langs
             index_object['languages'] = resource.serialize['languages']
         if json.get('category'):
+            old_category = resource.category #I'm assuming this is the category name as a string
+            resource_categories = get_all_resource_categories_as_strings()
             resource.category = categ
             index_object['category'] = categ.name
+            # Check if the old category is being used anywhere else
+            # in the Resources table
+            orphaned_category = Category.query \
+                .filter(Category.name.notin_(resource_categories)) \
+                .filter(Category.name == old_category) \
+                .first()
```

Comment:
Does this actually need to be a SQL query? Can we not just loop through the array of strings `resource_categories` and look for `old_category` and if it's not in there, we know that `old_category` is orphaned?

I haven't thought through it a whole lot, so if the answer is we do indeed need a SQL query then I'm willing to hear about why. But at first glance it looks like we don't

Reply:
Since `old_category` is a Category object, we can just call `old_category.delete()` on it if it's orphaned right? In that case, we wouldn't need the query you highlighted above. I can just tweak the following `if` statement and be done with it. I'll have another commit tonight.

Reply:
I think you delete with `db.session.delete(old_category)`

Reply:
Thanks for saying "I think" so I don't look as dumb for saying that.